File: 2014-10-CSharp-Fundamentals-1/2. Primitive Data Types and Variables/NullableTypes/NullableTypes.cs
using System;

class NullableTypes
{
    static void Main()
    {
        Console.WriteLine(
			new string('-', 40) +
			"\nExample with integer\n" +
			new string('-', 40));

		int? someInteger = null;
        //int someInteger = null;
        Console.WriteLine(
			"This is the integer with Null value -> " + someInteger);
        someInteger = 5;
        Console.WriteLine(
			"This is the integer with value 5 -> " + someInteger);

        
		Console.WriteLine(
			new string('-', 40) + 
			"\nExample with double\n" + 
			new string('-', 40));
        
        double? someDouble;
        //double someDouble;
        someDouble = null;
        Console.WriteLine(
			"This is the integer with Null value -> " + someDouble);
        someDouble = 2.5;
        Console.WriteLine(
			"This is the integer with value 5 -> " + someDouble);

		int? value = null;
		Console.WriteLine(value.GetValueOrDefault());
		// Console.WriteLine(value.Value); // This will cause an exception
	}
}

File: 2015-10-Java-Fundamentals-1/02. Data-types-and-variables/demos/NullableTypes.java
public class NullableTypes {
  public static void main(String[] args) {
    System.out.println("----------------------------------------");

    Integer someInteger = null;
    //int someInteger = null;
    System.out.printf("This is the integer with Null value -> %d%n",
                      someInteger);
    someInteger = 5;
    System.out.printf("This is the integer with value 5 -> %d%n",
                      someInteger);

    System.out.println("----------------------------------------");

    Double someDouble;
    //double someDouble;
    someDouble = null;
    System.out.printf("This is the double with Null value -> %f%n", 
                      someDouble);

    someDouble = 2.5;
    System.out.printf("This is the double with value 2.5 -> %f%n",
                      someDouble);

    //this will throw an NullPointerException
    Integer value = null;
    System.out.println(value.intValue());
  }
}
